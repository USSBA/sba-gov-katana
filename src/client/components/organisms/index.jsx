/* IMPORTS */

// miscellaenous organisms
import BusinessGuideTileCollection from "./business-guide-tile-collection/business-guide-tile-collection.jsx";
import CardCollection from "./card-collection/card-collection.jsx";
import DeveloperTester from "./developer-tester/developer-tester.jsx";
import DocumentCardCollection from "./document-card-collection/document-card-collection.jsx";
import FrontPageHero from "./front-page-hero/front-page-hero.jsx";
import FundingProgramsTileCollection from "./funding-programs-tile-collection/funding-programs-tile-collection.jsx";
import DisasterAlert from "./header-footer/disaster-alert/disaster-alert.jsx";
import NotificationBar from "./header-footer/notification-bar.jsx";
import Hero from "./hero/hero.jsx";
import RelatedDocumentCards from "./related-document-cards/related-document-cards.jsx";
import SearchBox from "./search-box/search-box.jsx";
import SectionNav from "./section-nav/section-nav.jsx";

// header-footer organisms
import Footer from "./header-footer/footer/footer.jsx";
import MobileNav from "./header-footer/mobile-nav/mobile-nav.jsx";
import Header from "./header-footer/header/header.jsx";

// homepage organisms
import Blog from "./homepage/blog/blog.jsx";
import FrontPageLady from "./homepage/front-page-lady/front-page-lady.jsx";
import HappeningNow from "./homepage/happening-now/happening-now.jsx";

// lender match organisms
// -- form
import AdditionalInfoForm from "./lender-match/form/additional-info.jsx";
import BusinessInfoForm from "./lender-match/form/business-info.jsx";
import ContactInfoForm from "./lender-match/form/contact-info.jsx";
import IndustryInfoForm from "./lender-match/form/industry-info.jsx";
import LoanInfo from "./lender-match/form/loan-info.jsx";
import ReviewSection from "./lender-match/form/review-page-helpers.jsx";
import ReviewSubmitInfoForm from "./lender-match/form/review-submit-info.jsx";
// -- landing page
import FindLendersIntro from "./lender-match/landing-page/find-lenders-intro.jsx";
import HelpfulQuestions from "./lender-match/landing-page/helpful-questions.jsx";
import {HowItWorksSection} from "./lender-match/landing-page/how-it-works.jsx";
import PreparationChecklist from "./lender-match/landing-page/preparation-checklist.jsx";
// -- success page
import ConfirmSection from "./lender-match/success-page/confirm-section.jsx";
import DynamicCounselingAndTools from "./lender-match/success-page/counseling-and-tools.jsx";
import CounselorMap from "./lender-match/success-page/counselor-map.jsx";
import EmailConfirmationInvalid from "./lender-match/success-page/email-confirmed-page.jsx";

// modal organisms
import LeaveSbaModal from "./modals/leave-sba-modal/leave-sba-modal.jsx";
import MobileSectionNavModal from "./modals/mobile-section-nav/mobile-section-nav.jsx";

/* EXPORTS */

// miscellaneous organisms
export {
	BusinessGuideTileCollection,
	CardCollection,
	DeveloperTester,
	DocumentCardCollection,
	FrontPageHero,
	FundingProgramsTileCollection,
	DisasterAlert,
	NotificationBar,
	Hero,
	RelatedDocumentCards,
	SearchBox,
	SectionNav
};

// header-footer organisms
export {
	Footer,
	MobileNav,
	Header
};

// homepage organisms
export {
	Blog,
	FrontPageLady,
	HappeningNow
};

// lender match organisms
export {
	AdditionalInfoForm,
	BusinessInfoForm,
	ContactInfoForm,
	IndustryInfoForm,
	LoanInfo,
	ReviewSection,
	ReviewSubmitInfoForm,
	FindLendersIntro,
	HelpfulQuestions,
	HowItWorksSection,
	PreparationChecklist,
	ConfirmSection,
	DynamicCounselingAndTools,
	CounselorMap,
	EmailConfirmationInvalid
};

// modal organisms
export {
	LeaveSbaModal,
	MobileSectionNavModal
};
