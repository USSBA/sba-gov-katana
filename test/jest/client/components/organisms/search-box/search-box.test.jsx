import React from 'react'
import renderer from 'react-test-renderer'
import sinon from 'sinon'
import { mount, shallow } from 'enzyme'
import { clone } from 'lodash'

import { SearchBox } from 'organisms'

jest.mock('client/services/client-config.js', function() {
  return {
    googleAnalytics: {
      enabled: false
    }
  }
})

describe('searchBox', () => {
  // has 1 h2 tag
  // has 1 search field
  // has 1 multiselect control
  // has 1 button
  // button reads "Submit"
  // fires an onClick event

  test('has 1 h2 tag', () => {
    const component = shallow(<SearchBox />)
    expect(component.find('h2')).toHaveLength(1)
  })

  test('has 1 TextInput Component', () => {
    const component = shallow(<SearchBox />)
    expect(component.find('TextInput')).toHaveLength(1)
  })

  test('has 3 Multiselect Component', () => {
    const component = shallow(<SearchBox />)
    expect(component.find('.multiSelect')).toHaveLength(3)
  })

  test('has 1 Submit Button Component', () => {
    const component = shallow(<SearchBox />)
    expect(component.find('Button')).toHaveLength(1)
  })

  test("Submit Button Component reads, 'Search'", () => {
    const mockText = 'Search'
    const component = shallow(<SearchBox />)
    const expectedText = component
      .find('Button')
      .first()
      .props().children
    expect(expectedText).toBe(mockText)
  })

  test('should render default sectionHeaderText and subtitleText, h2 and p respectively', () => {
    const component = renderer.create(<SearchBox />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree.children[0].children[0].type).toBe('h2')
    expect(tree.children[0].children[0].children[0]).toBe('Search documents, forms, and SOPs')
    expect(tree.children[0].children[1].type).toBe('p')
    expect(tree.children[0].children[1].children[0]).toBe('Search by title or document number')
  })

  test('should have default props for document type, program and document activity, and its default labels', () => {
    const component = shallow(<SearchBox />)
    expect(component).toMatchSnapshot()
    expect(component.instance().props.multiSelectDocumentTypeDefaultLabel).toBe('All document types')
    expect(component.instance().props.multiSelectProgramDefaultLabel).toBe('All programs')
    expect(component.instance().props.multiSelectDocumentActivityDefaultLabel).toBe('All document activity')
    expect(component.instance().props.documentType).toEqual([
      'SBA form',
      'SOP',
      'Policy Guidance',
      'TechNote',
      'Procedural notice',
      'Information notice',
      'Policy notice',
      'Support'
    ])
    expect(component.instance().props.program).toEqual([
      'SBIC',
      'Surety Bonds',
      '7(a)',
      'CDC/504',
      'Microlending',
      'HUBZone',
      'Disaster',
      '8(a)',
      'SBA operations',
      'Contracting',
      'Community Advantage'
    ])
    expect(component.instance().props.documentActivity).toEqual([
      'Authorization',
      'Servicing',
      'Closing',
      'Liquidation',
      'Litigation',
      'Guaranty purchase',
      'Licensing and organizational',
      'Credit and risk',
      'Investment and transactions',
      'Leverage commitments and draws',
      'Periodic reporting',
      'General',
      'Processing',
      'Secondary market'
    ])
  })

  test("should render documentType, program and documentActivity as 'All' as its default value and empty string for search term", () => {
    const component = mount(<SearchBox />)
    expect(component.state().searchTerm).toBe('')
    expect(component.state().selectedDocumentType).toBe('All')
    expect(component.state().selectedProgram).toBe('All')
    expect(component.state().selectedDocumentActivity).toBe('All')
  })

  test('should update state for document type, program, document acitivity and search term', () => {
    const props = {
      documentType: { value: 'TechNote' },
      program: { value: 'SBIC' },
      documentActivity: { value: 'Authorization' },
      searchTerm: { target: { value: 'newWord' } }
    }
    const component = shallow(<SearchBox />)
    expect(component).toMatchSnapshot()
    component.instance().handleChange(props.documentType)
    component.instance().handleChange(props.program)
    component.instance().handleChange(props.documentActivity)
    component.instance().updateSearchTerm(props.searchTerm)
    expect(component.instance().state.selectedDocumentType).toBe('TechNote')
    expect(component.instance().state.selectedProgram).toBe('SBIC')
    expect(component.instance().state.selectedDocumentActivity).toBe('Authorization')
    expect(component.instance().state.searchTerm).toBe('newWord')
  })

  test('should not update state for document type, program and document acitivity', () => {
    const props = {
      selectedLabel: { value: 'itis' }
    }
    const component = shallow(<SearchBox />)
    component.instance().handleChange(props.selectedLabel)
    expect(component).toMatchSnapshot()
    expect(component.instance().state.selectedDocumentType).not.toBe('itis')
    expect(component.instance().state.selectedProgram).not.toBe('itis')
    expect(component.instance().state.selectedDocumentActivity).not.toBe('itis')
  })

  test('should update default props for sectionHeaderText and subtitleText', () => {
    const testProps = {
      sectionHeaderText: 'new title',
      subtitleText: 'new subtitle'
    }
    const props = clone(testProps)
    const component = shallow(<SearchBox {...props} />)
    expect(component).toMatchSnapshot()
    expect(component.instance().props.sectionHeaderText).toBe('new title')
    expect(component.instance().props.subtitleText).toBe('new subtitle')
  })

  /* To Do: this test needs to be fleshed out more
	// for some reason 'preventDefault' is not stopping the submit call from executing/
	// This causes the following error:
	// ---
	// TypeError: Could not parse "/document/?search=&type=SBA form&program=8(a)&activity=All" as a URL
	// ---

	test("Submit Button fires 'submit' when onClick is invoked", () => {

		const preventDefault = jest.fn();
		const component = shallow(<SearchBox />);
		const instance = component.instance();
		const spy = sinon.spy(instance, "submit");
		instance.forceUpdate();
		component.find("LargeInversePrimaryButton").first().simulate("click", { preventDefault });

		sinon.assert.calledOnce(spy);

	});
	*/
})
