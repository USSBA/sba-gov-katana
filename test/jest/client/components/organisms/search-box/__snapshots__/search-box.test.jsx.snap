// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`searchBox should have default props for document type, program and document activity, and its default labels 1`] = `ShallowWrapper {}`;

exports[`searchBox should not update state for document type, program and document acitivity 1`] = `ShallowWrapper {}`;

exports[`searchBox should render default sectionHeaderText and subtitleText, h2 and p respectively 1`] = `
<div
  className="container"
>
  <div
    className="blueParagraph"
  >
    <h2>
      Search documents, forms, and SOPs
    </h2>
    <p>
      Search by title or document number
    </p>
    <div
      className="searchBox"
    >
      <div
        className="text-input textInput"
        data-testid="document-lookup-container"
        id="document-lookup-container"
      >
        <label
          className={null}
          data-testid="document-lookup-label"
          htmlFor="document-lookup"
        />
        <div
          className="container"
        >
          <input
            aria-labelledby="document-lookup"
            className="input"
            data-testid="document-lookup"
            id="document-lookup"
            inputMode="text"
            list=""
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyUp={[Function]}
            placeholder="Search by title or number"
            required={true}
            type="text"
          />
        </div>
      </div>
      <div
        className="searchIcon"
      >
        <i
          className="search fa fa-search"
        />
      </div>
    </div>
    <div
      className="multiSelect"
    >
      <div
        className="select"
        data-cy="documenttype-select"
        data-testid="documenttype-select"
      >
        <label
          htmlFor="documenttype-select"
        />
        <div
          className="Select has-value is-clearable is-searchable Select--single"
        >
          <input
            disabled={false}
            name="documenttype-select"
            type="hidden"
            value="All"
          />
          <div
            className="Select-control"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
          >
            <span
              className="Select-multi-value-wrapper"
              id="react-select-2--value"
            >
              <div
                className="Select-value"
              >
                <span
                  aria-selected="true"
                  className="Select-value-label"
                  id="react-select-2--value-item"
                  role="option"
                >
                  All document types
                </span>
              </div>
              <div
                className="Select-input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-activedescendant="react-select-2--value"
                  aria-expanded="false"
                  aria-haspopup="false"
                  aria-owns=""
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  role="combobox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "5px",
                    }
                  }
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </span>
            <span
              aria-label="Clear value"
              className="Select-clear-zone"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              title="Clear value"
            >
              <span />
            </span>
            <span
              className="Select-arrow-zone"
              onMouseDown={[Function]}
            >
              <i
                alt="drop-down arrow"
                className="fa fa-chevron-down"
              />
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="multiSelect"
    >
      <div
        className="select"
        data-cy="program-select"
        data-testid="program-select"
      >
        <label
          htmlFor="program-select"
        />
        <div
          className="Select has-value is-clearable is-searchable Select--single"
        >
          <input
            disabled={false}
            name="program-select"
            type="hidden"
            value="All"
          />
          <div
            className="Select-control"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
          >
            <span
              className="Select-multi-value-wrapper"
              id="react-select-3--value"
            >
              <div
                className="Select-value"
              >
                <span
                  aria-selected="true"
                  className="Select-value-label"
                  id="react-select-3--value-item"
                  role="option"
                >
                  All programs
                </span>
              </div>
              <div
                className="Select-input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-activedescendant="react-select-3--value"
                  aria-expanded="false"
                  aria-haspopup="false"
                  aria-owns=""
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  role="combobox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "5px",
                    }
                  }
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </span>
            <span
              aria-label="Clear value"
              className="Select-clear-zone"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              title="Clear value"
            >
              <span />
            </span>
            <span
              className="Select-arrow-zone"
              onMouseDown={[Function]}
            >
              <i
                alt="drop-down arrow"
                className="fa fa-chevron-down"
              />
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="multiSelect"
    >
      <div
        className="select"
        data-cy="documentactivity-select"
        data-testid="documentactivity-select"
      >
        <label
          htmlFor="documentactivity-select"
        />
        <div
          className="Select has-value is-clearable is-searchable Select--single"
        >
          <input
            disabled={false}
            name="documentactivity-select"
            type="hidden"
            value="All"
          />
          <div
            className="Select-control"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
          >
            <span
              className="Select-multi-value-wrapper"
              id="react-select-4--value"
            >
              <div
                className="Select-value"
              >
                <span
                  aria-selected="true"
                  className="Select-value-label"
                  id="react-select-4--value-item"
                  role="option"
                >
                  All document activity
                </span>
              </div>
              <div
                className="Select-input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-activedescendant="react-select-4--value"
                  aria-expanded="false"
                  aria-haspopup="false"
                  aria-owns=""
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  role="combobox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "5px",
                    }
                  }
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </span>
            <span
              aria-label="Clear value"
              className="Select-clear-zone"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              title="Clear value"
            >
              <span />
            </span>
            <span
              className="Select-arrow-zone"
              onMouseDown={[Function]}
            >
              <i
                alt="drop-down arrow"
                className="fa fa-chevron-down"
              />
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="clear"
    />
    <button
      className="button alternate large primary responsive spacing"
      data-testid="button"
      onClick={[Function]}
    >
      Search
    </button>
  </div>
</div>
`;

exports[`searchBox should update default props for sectionHeaderText and subtitleText 1`] = `ShallowWrapper {}`;

exports[`searchBox should update state for document type, program, document acitivity and search term 1`] = `ShallowWrapper {}`;
